ivy/utils/exceptions.py

```diff
@@ -177,6 +182,9 @@ def __init__(self, *messages, include_backend=False):
 
 
 def handle_exceptions(fn: Callable) -> Callable:
+
+    buffer = io.StringIO()
```

Comment:
Could you please refactor this so that we wouldn't be creating the buffer twice? We seem to also instantiate at line [`44`](https://github.com/unifyai/ivy/pull/15862/files#diff-97b4bf909646477810e1e9162329746a1fb59a7938988d918ee1873e9d872cdeR44)

Reply:
I think, there's some misunderstanding. Handling custom hook is independent from handling exception. Like they can't access same buffer variable. So there must be a separate buffer for each. let me know if I'm missing something.
Thanks!

Reply:
Oh okay, I was thinking if there could be a way, but no worries if not!